specs/ChainSpec/Consensus.md

```diff
@@ -0,0 +1,147 @@
+# Consensus
+
+The fields in the Block header relevant to the consensus process are:
+
+```rust
+struct BlockHeader {
+    ...
+    prev_hash: BlockHash,
+    height: BlockHeight,
+    epoch_id: EpochId,
+    last_final_block_hash: BlockHash,
+    approvals: Vec<Approval>
+    ...
+}
+```
+
+Each block belongs to a particular epoch, and has a particular height. The highest height block known to a particular node is its head.
+
+Each epoch has a set of block producers who are assigned to produce blocks in the epoch, and has an assigned block producer for each height in the epoch. The block producers and the assignments in a particular epoch are known from the first block of the *preceding* epoch.
```

Comment:
link to epoch manager for more details of the assignment? or for high level API needed?

Reply:
I don't think we have epoch manager spec yet?
As we have other sections written, we can add the cross links.